### PR TITLE
Misc Improvements

### PR DESCRIPTION
**Pull request by jd-lara**
Addresses comments in #9

### CODE REVIEW

File: src/base/simulation.jl [hunk 1]
```diff
@@ -21,6 +48,25 @@ function Simulation(
 )
     check_folder(simulation_folder)
     simulation_system = deepcopy(system)
```
Comment (rodrigomha) [inline src/base/simulation.jl:50]: Shouldn't we want to avoid doing this all the time? Maybe add an extra keyword?

Comment (jd-lara) [reply src/base/simulation.jl:50]: `Simulation vs Simulation!`

Comment (daniel-thom) [reply src/base/simulation.jl:50]: It's odd that you might end up with two serializations of the PSY.System.

Comment (jd-lara) [reply src/base/simulation.jl:50]: We can't assume that the original system came from a deserialized json file.

**Pull request merged**